study-builder/studies/atcp/study.conf

```diff
@@ -0,0 +1,1192 @@
+      "dispatchToHousekeeping": false,
+      "order": 99
+    },
+        {
+      "trigger": {
+        "type": "ACTIVITY_STATUS",
+        "activityCode": "GENOME_STUDY",
+        "statusType": "CREATED"
+      },
+      "action": {
+        "type": "COPY_ANSWER",
+        "copyConfigPairs": [
+          {
+            "source": {
+              "type": "ANSWER",
+              "questionStableId": "REGISTRATION_STREET_ADDRESS"
+            },
+            "target": {
+              "type": "ANSWER",
+              "questionStableId": "SHIPPING_ADDRESS"
+            }
+          },
+          {
+            "source": {
+              "type": "ANSWER",
+              "questionStableId": "REGISTRATION_GENDER"
+            },
+            "target": {
+              "type": "ANSWER",
+              "questionStableId": "PARTICIPANT_GENDER"
+            }
+          }
+        ]
+      },
+      "preconditionExpr": """!user.studies["atcp"].forms["GENOME_STUDY"].instances[latest].hasPreviousInstance()""",
+      "cancelExpr": null,
+      "maxOccurrencesPerUser": 1,
+      "delaySeconds": null,
+      "dispatchToHousekeeping": false,
+      "order": 100
+    }
+    # populate profile events
+    {
+      "trigger": {
+        "type": "ACTIVITY_STATUS",
+        "activityCode": "CONSENT",
+        "statusType": "COMPLETE"
+      },
+      "action": {
+        "type": "COPY_ANSWER",
+        "copyConfigPairs": [
+          {
+            "source": {
+              "type": "ANSWER",
+              "questionStableId": "CONSENT_DOB"
+            },
+            "target": {
+              "type": "PARTICIPANT_PROFILE_BIRTH_DATE"
+            }
+          }
+        ]
+      },
+      "dispatchToHousekeeping": false,
+      "order": 1
+    },
+
+    ## review and submission email
+    {
+      "trigger": {
+        "type": "ACTIVITY_STATUS",
+        "activityCode": "REVIEW_AND_SUBMISSION",
+        "statusType": "COMPLETE"
+      },
+      "action": {
+        "type": "STUDY_EMAIL",
+        "templates": [
+          {
+            "emailTemplate": ${emails.selfEnrollmentCompleted},
+            "language": "en",
+            "is_dynamic": false
+          }
+        ],
+        "pdfAttachments": []
+      },
+      "maxOccurrencesPerUser": 1,
+      "dispatchToHousekeeping": true,
+      "order": 1
+    },
+
+    {
+      "trigger": {
+        "type": "ACTIVITY_STATUS",
+        "activityCode": "REGISTRATION",
+        "statusType": "COMPLETE"
+      },
+      "action": {
+        "type": "STUDY_EMAIL",
+        "templates": [
+          {
+            "emailTemplate": ${emails.registrationUpdated},
+            "language": "en",
+            "is_dynamic": false
+          }
+        ],
+        "pdfAttachments": []
+      },
+      "maxOccurrencesPerUser": null,
+      "dispatchToHousekeeping": true,
+      "order": 1,
+      "preconditionExpr": """user.studies["atcp"].forms["REGISTRATION"].instances[latest].hasPreviousInstance()"""
```

Comment:
What happens if there is no previous instance? Since `dispatchToHousekeeping` is true, the event will be evaluated by Housekeeping. And if `preconditionExpr` is false, Housekeeping will continue to re-evaluate this event indefinitely. Will `preconditionExpr` ever evaluate to true? I recommend changing this event to use `cancelExpr` instead, so when `cancelExpr` evaluates to true Housekeeping will drop the event and won't continue to re-evaluate the event.

Reply:
Switched to the cancelExpr